src/Momento.Sdk.Incubating/Responses/CacheListFetchResponse.cs

```diff
@@ -4,43 +4,81 @@
 using Google.Protobuf;
 using Google.Protobuf.Collections;
 using Momento.Protos.CacheClient;
+using Momento.Sdk.Exceptions;
 using Momento.Sdk.Responses;
 
 namespace Momento.Sdk.Incubating.Responses;
 
-public class CacheListFetchResponse
+public abstract class CacheListFetchResponse
 {
-    public CacheGetStatus Status { get; private set; }
-    private readonly RepeatedField<ByteString>? values;
-    private readonly Lazy<List<byte[]>?> _byteArrayList;
-    private readonly Lazy<List<string>?> _stringList;
-
-    public CacheListFetchResponse(_ListFetchResponse response)
+    public class Hit : CacheListFetchResponse
     {
-        Status = (response.ListCase == _ListFetchResponse.ListOneofCase.Found) ? CacheGetStatus.HIT : CacheGetStatus.MISS;
-        values = (Status == CacheGetStatus.HIT) ? response.Found.Values : null;
+        protected readonly RepeatedField<ByteString>? values;
+        protected readonly Lazy<List<byte[]>?> _byteArrayList;
+        protected readonly Lazy<List<string>?> _stringList;
 
-        _byteArrayList = new(() =>
+        public Hit(_ListFetchResponse response)
         {
-            if (values == null)
+            values = response.Found.Values;
+            _byteArrayList = new(() =>
             {
-                return null;
-            }
+                return new List<byte[]>(values.Select(v => v.ToByteArray()));
+            });
+
+            _stringList = new(() =>
+            {
+                return new List<string>(values.Select(v => v.ToStringUtf8()));
+            });
+        }
+
+        public List<byte[]>? ByteArrayList { get => _byteArrayList.Value; }
 
-            return new List<byte[]>(values.Select(v => v.ToByteArray()));
-        });
+        public List<string>? StringList() => _stringList.Value;
+    }
 
-        _stringList = new(() =>
+    public class Miss : CacheListFetchResponse
+    {
+        protected readonly Lazy<List<byte[]>?> _byteArrayList;
+        protected readonly Lazy<List<string>?> _stringList;
+        public Miss()
         {
-            if (values == null)
+            _byteArrayList = new(() =>
+            {
+                return null;
+            });
+
+            _stringList = new(() =>
             {
                 return null;
-            }
-            return new List<string>(values.Select(v => v.ToStringUtf8()));
-        });
+            });
+        }
+        public List<byte[]>? ByteArrayList { get => _byteArrayList.Value; }
+
+        public List<string>? StringList() => _stringList.Value;
```

Comment:
This should not have any properties.

Reply:
Based on this test, we want null for Miss: https://github.com/momentohq/client-sdk-dotnet-incubating/blob/c0c0705006ad924e65b4ad9868df245ca67ffa9b/tests/Integration/Momento.Sdk.Incubating.Tests/ListTest.cs#L474

Again I'll defer to @malandis